hefquin-vocabs/feddesc.ttl

```diff
@@ -0,0 +1,90 @@
+@prefix rdf:    <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
+@prefix rdfs:   <http://www.w3.org/2000/01/rdf-schema#> .
+@prefix owl:    <http://www.w3.org/2002/07/owl#> .
+@prefix xsd:    <http://www.w3.org/2001/XMLSchema#> .
+@prefix dc:     <http://purl.org/dc/elements/1.1/> .
+@prefix fd:     <http://w3id.org/hefquin/feddesc#> .
+
+<http://w3id.org/hefquin/feddesc>
+    rdf:type owl:Ontology ;
+    dc:creator       "Olaf Hartig",
+                     "Robin Keskisärkkä" ;
+    dc:date          "2025-05-05" ;
+    dc:format        "RDF" ;
+    dc:identifier    <http://w3id.org/hefquin/feddesc> ;
+    dc:title         "Federation Description Vocabulary for HeFQUIN"@en ;
+    rdfs:label       "Federation Description Vocabulary for HeFQUIN"@en ;
+    dc:description   "An RDF vocabulary for describing members of a federation to be considered for querying by the HeFQUIN engine."@en ;
+    rdfs:comment     "An RDF vocabulary for describing members of a federation to be considered for querying by the HeFQUIN engine."@en .
+
+### ===== CLASS DECLARATIONS =====
+
+fd:FederationMember a owl:Class ;
+    rdfs:label "Federation Member"@en ;
+    rdfs:comment "An entity that is part of the federation and provides data via a specific interface."@en .
+
+fd:EndpointInterface a owl:Class ;
+    rdfs:label "Endpoint Interface"@en ;
+    rdfs:comment "An endpoint interface is an interface to request a dataset or fragment (i.e., a specific subset) of a dataset from a federation member. This class is the general superclass for all types of interfaces through which a federation member can be queried, including SPARQL interfaces, and fragment interfaces. It should not be instantiated directly in federation descriptions, instead, concrete subclasses should be used to specify the type of interface."@en .
+
+fd:SPARQLEndpointInterface a owl:Class ;
+    rdfs:subClassOf fd:EndpointInterface ;
+    rdfs:label "SPARQL Endpoint Interface"@en ;
+    rdfs:comment "An interface to access a federation member via the SPARQL protocol."@en .
+
+fd:FragmentInterface a owl:Class ;
+    rdfs:subClassOf fd:EndpointInterface ;
```

Comment:
Sorry for bikeshedding but that's not how I see it. An endpoint interface is an interface that has a single endpoint whereas a fragment interface can have an unbounded number of endpoints, one per fragment. Also, with this way of modeling (in combination with the fact that the `rdfs:domain` of `fd:endpointAddress` is `fd:EndpointInterface`), it becomes possible that fragment interfaces have an endpoint address. The fix is to introduce `fd:Interface` as top-level superclass, with `fd:EndpointInterface` and `fd:FragmentInterface` as subclasses.

Reply:
No worries, I see your point. It would not make sense to be able to specify `fd:endpointAddress` for fragment interfaces, and this should be clear from the vocabulary specification.

I've added a `fd:Interface` as the the top-level superclass for interfaces and essentially. This now has the more general description, whereas `fd:EndpointInterface` explicitly states in the comment that it is intended for a single endpoint. I also change the range of `fd:interface` to the new `fd:Interface` class.

Reply:
Btw, the main branch had deviated quite a bit from this feature branch due to the recent PRs merged, so i rebased this branch.